mypyc/emitfunc.py

```diff
@@ -316,7 +316,8 @@ def visit_method_call(self, op: MethodCall) -> None:
         # Can we call the method directly, bypassing vtable?
         is_direct = False
         if not class_ir.is_trait:
-            if not any(name in subc.methods for subc in class_ir.subclasses()):
+            if all(subc.get_method(name) is method
+                    for subc in class_ir.concrete_subclasses()):
```

Comment:
Why the switch to concrete_subclasses?

Reply:
I just figured every instance has a concrete subclass as a type